### PR TITLE
[JENKINS-25276] NestedView rename/delete support

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-25276](https://issues.jenkins-ci.org/browse/JENKINS-25276)

@reviewbybees but untested; @lvotypko may want to pick this up to confirm with automated and/or manual testing.

## Changelog Entry

* Jobs do not longer disappear from NestedView lists  after renaming/delete

### CODE REVIEW

File: core/src/main/java/hudson/model/ListView.java [hunk 1]
```diff
@@ -486,19 +487,20 @@ private void renameViewItem(String oldFullName, String newFullName, ViewGroup vg
...
+            }
+        }
+        private void deleted(ViewGroup vg, Item item) {
+            for (View v : vg.getViews()) {
+                if (v instanceof ListView) {
+                    deleteViewItem(item, vg, (ListView) v);
+                }
+                if (v instanceof ViewGroup) {
+                    deleted((ViewGroup) v, item);
                 }
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/ListView.java]: I don't like such <code>if</code>s without an action for the unsupported action (e.g. logging). Probably it's OK for this plugin, because such patterns are everywhere

**Comment by oleg-nenashev** [comment]
IMO it needs some logging for <code>else</code> cases before merging.

File: core/src/main/java/hudson/model/ListView.java (hunk 1 above)
Comment (jglick) [inline core/src/main/java/hudson/model/ListView.java]: What are you proposing to log? That there is something that is neither a `ListView` nor a `NestedView`? That is not an error condition here, because this listener is part of `ListView`. If some other (non-nested) view needs similar logic, it needs its own listener.

Comment (jglick) [inline core/src/main/java/hudson/model/ListView.java]: Anyway the comment seems out of scope for this PR, which merely adds `NestedView` support to an existing `instanceof` check.

Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/ListView.java]: Yes, it's out of the scope. I'm OK with the current implementation

I'm mostly aware that such <code>if</code>s without else blocks increase risks of hidden errors if somebody adds a new type, which includes ListView somehow (Example: Sectioned View Plugin). BTW it's another story

**Comment by oleg-nenashev** [comment]
After digging into the code I'm OK to leave the PR as is => :bee: 

@jglick , are you waiting for tests from @lvotypko ?

**Comment by jglick** [comment]
> are you waiting for tests from @lvotypko ?

Yes.

**Comment by jglick** [comment]
@lvotypko ping

**Review by olivergondza — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
I see no issues with the code. Tests would be still useful, but the change is straightforward enough

**Comment by oleg-nenashev** [comment]
@reviewbybees done I'd guess

**Comment by jglick** [comment]
> Tests would be still useful

Already present in plugin PR, q.v. I just tested and confirmed that the test fails against 2.86 but passes against this PR.

**Comment by daniel-beck** [comment]
> Jobs do not longer disappear from NestedView lists after renaming/**delete**

Huh?
